### DISCUSSION TITLE
Android App keeps stopping

### DISCUSSION THREAD

**Discussion by ejsilver26** [chuot/rdio-scanner#84] [Q&A]
I am using the android app for rdio. The app keeps stopping at random times. I have searched through all the setting that I can find to keep it running, even allowing it to run when the "Do Not Disturb" is on. I have a Samsung phone, run android 11, and am running rdio 1.3.2. Thanks.

**Comment by noamlivne** [comment]
I also have this issue. But I think it is related to Android's aggressive battery saver and not to this app specifically. Turning off the Battery Restrictions for the app, probably helps but does not solve the issue. I would guess that in older versions of Android the app will close less or not close at all.

**Reply by chuot** [reply]
This has been tested on a S9+ (SM-G965W) with Android 10 (One UI 2.5) and it does work. I have no other battery optimizer installed, but if I'm not turning off the Special Access / Optimize Battery Usage for Rdio Scanner, which is on by default,  it get killed for sure. That is why on Android I'm telling new users to turn that setting off for the Rdio Scanner app only.

**Comment by ejsilver26** [comment]
My issue could be WIFI/Bluetooth related. I just got home and was listening to Rdio in the car. I had it going through Bluetooth and was using my data. When I got home I disconnected the Bluetooth and the phone logged onto the home WIFI. After not hearing anything, I looked at the phone and opened the app. It opened, went blank for a second, then opened up, but was turned off. I turned it back on and it seems to be staying on.

**Reply by TPSSCW** [reply]
I'm having the same issue on a s22 running version 1.9.2...I really love the app but sure wish it would stay running

**Reply by noamlivne** [reply]
I don't have that phone, but perhaps there are some Settings that can be enabled such as no battery restrictions, locking the process as not to restart, auto start, etc.
Another option would be to use the built-in Samsung browser (and not the Android app) with the Rdio Scanner IP address. Perhaps that will work smoother.
